pkg/util/globalerror/errors.go

```diff
@@ -51,21 +51,17 @@ const (
 	MaxQueryLength       ID = "max-query-length"
 	RequestRateLimited   ID = "tenant-max-request-rate"
 	IngestionRateLimited ID = "tenant-max-ingestion-rate"
-	TooManyHAClusters    ID = "too-many-ha-clusters"
+	TooManyHAClusters    ID = "tenant-too-many-ha-clusters"
```

Comment:
This constant is also used to generate the `reason` label value for the metric:
https://github.com/grafana/mimir/blob/29142fc7ae465e31478483d818048070143ee688/pkg/util/validation/validate.go#L61

Changing this constant value, we're going to change the label value too, which is a breaking change we don't want to do (it's annoying for users). I suggest to change `ReasonTooManyHAClusters` to an hardcoded value which is equal to the previous value, that is:

```
ReasonTooManyHAClusters = "too_many_ha_clusters"
```